geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/NullRedisSortedSet.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package org.apache.geode.redis.internal.data;
+
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.geode.cache.Region;
+import org.apache.geode.redis.internal.executor.sortedset.SortedSetOptions;
+
+class NullRedisSortedSet extends RedisSortedSet {
+
+  NullRedisSortedSet() {
+    super(new ArrayList<>());
+  }
+
+  @Override
+  public boolean isNull() {
+    return true;
+  }
+
+  @Override
+  long zadd(Region<RedisKey, RedisData> region, RedisKey key, List<byte[]> membersToAdd,
+      SortedSetOptions options) {
+    if (options.isXX()) {
+      return 0;
+    }
+    region.create(key, new RedisSortedSet(membersToAdd));
+    return membersToAdd.size() / 2;
+  }
```

Comment:
I haven't  re-reviewed the whole PR yet, but Donal reminded me of a test scenario that currently fails... Right now if we do the following:

```
zadd green 1 a 2 a 3 a
```

Redis returns 1, but we return 3 since we're not actually verifying which members were added here vs. updated.

Reply:
That's gonna be fun, as we'll need to add checks to NullRedisSet, too. But good catch.

Reply:
I'm working on a bunch of these changes. I've added a method, "getSortedSetSize()" that returns "members.size()". In NullRedisSet, I return that for zadd: we went from 0 to whatever was added.

For zadd on an existing sorted set, I get the count from before we add anything, then at the end return "getSortedSetSize() - initialSize". If anything got updated but not added, it won't be counted. We shouldn't ever return negative, because there's no way zadd will remove anything.

Passes the test I put together for this case. I'm actually going to need two tests - one that hits NullRedisSortedSet and one that hits RedisSortedSet...

Reply:
My first stab at fixing this is up. Gonna do more cleanup, especially combining with other PR comments.

Reply:
Alright, a bit more cleanup finished.

Reply:
Should be cleaner now.